third-party/realdds/include/realdds/dds-device.h

```diff
@@ -50,13 +50,13 @@ class dds_device
 
     size_t num_of_sensors() const;
 
-    size_t foreach_sensor( std::function< void( size_t sensor_index, const std::string & name ) > fn ) const;
+    size_t foreach_sensor( std::function< void( const std::string & name ) > fn ) const;
 
-    size_t foreach_video_profile( size_t sensor_index, std::function< void( const rs2_video_stream& profile, bool def_prof ) > fn ) const;
-    size_t foreach_motion_profile( size_t sensor_index, std::function< void( const rs2_motion_stream& profile, bool def_prof ) > fn ) const;
+    size_t foreach_video_profile( std::string group_name, std::function< void( const rs2_video_stream& profile, bool def_prof ) > fn ) const;
```

Comment:
Please pass strings by const &.

Reply:
Added foreach_video_profile_in_group function

Reply:
Why? Is there a need?
If we're talking in terms of streams, why not `foreach_stream_profile`?

Reply:
dds_device now have a foreach_stream function and dds_stream has a foreach_profile function